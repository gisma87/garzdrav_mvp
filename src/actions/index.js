// ставим ошибку
const setError = (error) => {
  return {
    type: 'FETCH_FAILURE',
    payload: error
  }
}

//  очищаем ошибку
const clearError = () => {
  return {
    type: 'CLEAR_ERROR'
  }
}

// ставим загрузку
const loadingTrue = () => {
  return {
    type: 'LOADING'
  }
}

// очищаем массив CartItems
const delCartItems = () => {
  return {
    type: 'DEL_CART_ITEM'
  }
}


// собираем массивы cartItems и retailsArr
const setCartItems = (cartItems) => {
  return {
    type: 'SET_CART_ITEMS',
    payload: cartItems
  }
}

// серия запросов подробной информации о товаре из списка корзины - по IDproduct и IDcity.
// Формируется массив cartItems - список товаров со списком аптек в нём, где этот товар есть.
// Из массива cartItems формируется массив retailsArr - список аптек, со списком товаров из корзины имеющихся в этой аптеке.
const fetchCartItems = (city = null) => (dispatch, getState, apiService) => {
  const {cart, isCity} = getState()
  const cityId = city || isCity.guid
  dispatch(clearError())
  dispatch(delCartItems())

  if (cart.length > 0) {
    dispatch(loadingTrue())
    const arrFetch = cart.map(product => {
      return apiService.getProductInfo(product.itemId, cityId)
    })
    Promise.all([...arrFetch])
      .then(allResponses => dispatch(setCartItems(allResponses)))
      .catch(allError => dispatch(setError(allError)))
  }
}

// открывает/закрывает popup подтверждения города
const onPopupLocation = (boolean) => {
  return {
    type: 'ON_POPUP_LOCATION',
    payload: boolean
  }
}


// устанавливает объект текущего города и записывает его в LocalStorage
const setIsCity = (isCity) => (dispatch) => {
  dispatch(loadingTrue())
  const item = [isCity]
  console.log('Город: ', item);
  localStorage.setItem("city", JSON.stringify(item))
  // устанавливаем город
  dispatch({
    type: 'SET_CITY',
    payload: isCity
  })
  // запрашиваем инф. о товарах в корзине по новому городу
  dispatch(fetchCartItems(isCity.guid))
}

// запрос списка городов
const fetchCities = () => async (dispatch, getState, apiService) => {
  try {
    dispatch(loadingTrue())
    const response = await apiService.getCities()
    dispatch({type: 'FETCH_CITIES_SUCCESS', payload: response})
    dispatch(loadingTrue())

    // если в localStorage есть город - устанавливаем его
    if (localStorage.getItem("city")) {
      const cityItem = JSON.parse(localStorage.getItem("city"))[0]
      dispatch(setIsCity(cityItem))
    } else {

      // иначе определяем город по IP, устанавливаем его и открываем popup подтверждения выбранного города
      const {city, ip} = await apiService.getUserCity()
      console.log(city, ip);
      const cityItem = response.find(item => city === item.title)
      dispatch(setIsCity(cityItem))
      dispatch(onPopupLocation(true))
    }
  } catch (e) {
    dispatch(setError(e))
  }
}

// Список торговых точек в городе
const fetchRetailsCity = () => async (dispatch, getState, apiService) => {
  try {
    const cityId = getState().isCity.guid
    dispatch(loadingTrue())
    const response = await apiService.getRetailsCity(cityId)
    dispatch({type: 'FETCH_RETAILS_CITY_SUCCESS', payload: response})
  } catch (e) {
    dispatch(setError(e))
  }
}

const addedToCart = (ItemId) => {
  return {
    type: 'ITEM_ADDED_TO_CART',
    payload: ItemId
  }
}

const itemRemovedFromCart = (ItemId) => {
  return {
    type: 'ITEM_REMOVED_FROM_CART',
    payload: ItemId
  }
}

const allItemRemovedFromCart = (ItemId) => {
  return {
    type: 'ALL_ITEM_REMOVED_FROM_CART',
    payload: ItemId
  }
}

// все аптеки города
const retailsCityLoaded = (retailsCity) => {
  return {
    type: 'FETCH_RETAILS_CITY_SUCCESS',
    payload: retailsCity
  }
}

// перезаписываем корзину - обычно из значения localStorage
const rewriteCart = (cart) => {
  return {
    type: 'REWRITE_CART',
    payload: cart
  }
}

const addedToFavorits = (item) => {
  return {
    type: 'ADDED_TO_FAVORITES',
    payload: item
  }
}

// список позиций из поискового запроса
const ProductsFromSearchLoaded = (products, productSearch) => {
  products.productSearch = productSearch
  return {
    type: 'FETCH_PRODUCTS_FROM_SEARCH_SUCCESS',
    payload: products
  }
}

// поисковый запрос порционно с указанием количества элементов и страницы
function getProductsFromSearchLimit(productSearch, quantity, page, methodSort) {
  console.log('methodSort', methodSort)
  return async (dispatch, getState, apiService) => {
    const cityId = getState().isCity.guid
    dispatch(loadingTrue())
    try {
      const response = await apiService.getProductsFromSearchLimit(productSearch, cityId, quantity, page, methodSort)
      dispatch(ProductsFromSearchLoaded(response, productSearch))
    } catch (e) {
      dispatch(setError(e))
    }
  }
}


// дополнительная(подробная) информация о продукте
const fetchProductInfo = (productId, cityId) => {
  return async (dispatch, getState, apiService) => {
    dispatch(loadingTrue())
    try {
      const response = await apiService.getProductInfo(productId, cityId)
      dispatch(loadingProductInfo(response))
    } catch (e) {
      dispatch(setError(e))
    }
  }
}

// устанавливаем доп.инфу о товаре - который смотрим на страницу CardPage
const loadingProductInfo = (product) => {
  return {
    type: 'LOADING_PRODUCT_INFO',
    product
  }
}

const onSelectRetail = (id) => {
  return {
    type: 'ON_SELECT_RETAIL',
    payload: id
  }
}

const clearCart = () => {
  localStorage.removeItem("cart")
  return {
    type: 'CLEAR_CART'
  }
}

// запрос информации о пользователе по TOKEN из LocalStorage
const fetchUserData = () => async (dispatch, getState, apiService) => {
  const accessToken = getState().TOKEN.accessToken || JSON.parse(localStorage.getItem('TOKEN')).accessToken
  dispatch(loadingTrue())
  try {
    const response = await apiService.getUserData(accessToken)
    dispatch({type: 'USER_DATA', payload: response})
  } catch (e) {
    dispatch(setError(e))
  }
}

// POST запрос refreshTOKEN
const refreshAuthentication = () => async (dispatch, getState, apiService) => {
  const TOKEN = getState().TOKEN || JSON.parse(localStorage.getItem('TOKEN'))
  dispatch(loadingTrue())
  try {
    const response = await apiService.refreshToken(TOKEN)
    dispatch({type: 'TOKEN', payload: response})
    localStorage.setItem('TOKEN', JSON.stringify(response))
    console.log('refresh_TOKEN')
  } catch (e) {
    dispatch(setError(e))
    dispatch(logout())
  }
}

// POST запрос TOKEN
const authentication = () => async (dispatch, getState, apiService) => {
  dispatch(loadingTrue())
  try {
    const response = await apiService.authentication()
    dispatch({type: 'TOKEN', payload: response})
    localStorage.setItem('TOKEN', JSON.stringify(response))
    console.log('access_TOKEN')
    dispatch(fetchUserData(response.accessToken))
  } catch (e) {
    dispatch(setError(e))
  }
}


function logout() {
  localStorage.removeItem("TOKEN")
  return {
    type: 'LOGOUT'
  }
}

const setCatalog = () => async (dispatch, getState, apiService) => {
  dispatch(loadingTrue())
  try {
    const response = await apiService.buildCatalog()
    dispatch({
      type: 'SET_CATALOG',
      payload: response
    })
  } catch (e) {
    dispatch(setError(e))
  }
}

function setActiveCategory(categoryItem) {
  return {
    type: 'SET_ACTIVE_CATEGORY',
    payload: categoryItem
  }
}

const setProductsToCategory = (categoryId) => async (dispatch, getState, apiService) => {
  dispatch(loadingTrue())
  try {
    const response = await apiService.getProductToCategory(getState().isCity.guid, categoryId)
    dispatch({
      type: 'SET_PRODUCTS_TO_CATEGORY',
      payload: response
    })
  } catch (e) {
    dispatch(setError(e))
  }
}

const setSales = () => async (dispatch, getState, apiService) => {
  dispatch(loadingTrue())
  try {
    const response = await apiService.getSales(getState().TOKEN.accessToken)
    dispatch({
      type: 'SET_SALES',
      payload: response
    })
  } catch (e) {
    dispatch(setError(e))
  }
}


export {
  setSales,
  setProductsToCategory,
  setActiveCategory,
  setCatalog,
  logout,
  fetchUserData,
  refreshAuthentication,
  authentication,
  onPopupLocation,
  setCartItems,
  clearCart,
  setError,
  fetchCities,
  setIsCity,
  retailsCityLoaded,
  addedToCart,
  itemRemovedFromCart,
  allItemRemovedFromCart,
  rewriteCart,
  addedToFavorits,
  loadingTrue,
  ProductsFromSearchLoaded,
  fetchProductInfo,
  fetchCartItems,
  onSelectRetail,
  fetchRetailsCity,
  getProductsFromSearchLimit
}