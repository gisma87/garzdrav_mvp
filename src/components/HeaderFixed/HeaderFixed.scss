.HeaderFixed {
  background: #1f5377;
  min-height: 64px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  padding: 0 15px;
  font-family: bold, Helvetica, Arial, sans-serif;
  color: #fff;
  font-size: 15px;
  box-shadow: 0 10px 10px 0 rgba(0, 0, 0, .2);

  &__leftBlock {
    display: flex;
    align-items: center;
    justify-content: space-between;
    max-width: 350px;
    width: 100%;
  }

  &__catalog {
    position: relative;
  }

  //&__logo {
  //  height: 50px;
  //  width: 120px;
  //  display: flex;
  //  align-items: center;
  //  background-size: contain;
  //  justify-content: flex-start;
  //  cursor: pointer;
  //  //background-color: #747474;
  //  background-image: url("../../img/logo/iconALsmall.png");
  //  background-repeat: no-repeat;
  //}

  &__rightblock {
    display: flex;
    justify-content: flex-end;
    align-items: center;
    min-height: 50px;
    min-width: 300px;
    padding: 0 0 0 20px;
  }

  &__cartImgBox {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 35px;
    height: 35px;
    border-radius: 50%;
    background: #95c11f;
  }

  &__cartImg {
    transform: translateY(-1px);
    max-width: 30px;
    max-height: 22px;
    object-fit: contain;
    display: block;
  }

  &__cart {
    margin-left: 15px;
    background: #fff;
    border-radius: 25px;
    padding: 3px 17px 3px 3px;
    display: flex;
    align-items: center;
  }

  &__cartCount {
    transform: translateY(1px);
    font-family: bold, Helvetica, Arial, sans-serif;
    color: #1f5377;
    margin-left: 8px;
    font-size: 1.6rem;
    letter-spacing: -3px;
  }

  &__cartText {
    margin-left: 10px;
    color: #333;
  }

  &__logIn {
    color: #000;
    cursor: pointer;
    border: 1px solid #d9d9d9;
    background-color: #fff;
    padding: 10px 15px;
    border-radius: 8px;
    outline: none;

    &:hover {
      background-image: linear-gradient(to bottom, #ffbc0b, #ff7400);
      box-shadow: inset 0 -2px 0 0 rgba(0, 0, 0, 0.2);
      color: #fff;
      border-color: transparent;
      font-weight: bold;
    }

    &:active {
      background-image: linear-gradient(to top, #ffbc0b, #ff7400);
      box-shadow: inset 0 2px 0 0 rgba(0, 0, 0, 0.2);
    }
  }
}

.HeaderFixed--active {
  transition: all .1s;
  position: fixed;
  justify-content: center;
  width: 100%;
  z-index: 3;
}

.HeaderFixed__wrapper {
  box-shadow: none;
  justify-content: space-between;
}