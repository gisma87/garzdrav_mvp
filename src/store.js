import {applyMiddleware, compose, createStore} from "redux";
import reducer from "./reducers";
import thunk from 'redux-thunk';
import apiService from "./service/ApiService";

// это нужно в dev версии для плагина браузера React Developer Tools
const composeEnhancers =
  typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
    : compose;

// третьим параметром thunk в actions будет отдавать apiService
const enhancer = composeEnhancers(applyMiddleware(thunk.withExtraArgument(apiService)));

const store = createStore(reducer, enhancer)

export default store;