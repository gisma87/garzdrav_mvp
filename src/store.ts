import {applyMiddleware, compose, createStore} from "redux";
import reducer from "./reducers";
import thunk from 'redux-thunk';
import apiService from "./service/ApiService";

type ReducerType = typeof reducer;
export type StateType = ReturnType<ReducerType>

// это нужно в dev версии для плагина браузера React Developer Tools
const composeEnhancers =
  typeof window === 'object' && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
    : compose;

// третьим параметром thunk в actions будет отдавать apiService
const enhancer = composeEnhancers(applyMiddleware(thunk.withExtraArgument(apiService)));

const store = createStore(reducer, enhancer)

export default store;