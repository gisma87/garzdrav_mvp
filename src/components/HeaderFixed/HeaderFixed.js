import React, {useEffect, useState} from "react"
import './HeaderFixed.scss'
import {NavLink, withRouter} from "react-router-dom";
import SearchPanel from "../SearchPanel";
import iconCart from '../../img/icon/cartIconSmall.png'
import ButtonTopScroll from "../UI/ButtonTopScroll";
import {rewriteCart} from "../../actions";
import {connect} from "react-redux";
import PopupLogin from "../PopupLogin";
import Burger from "../UI/Burger/Burger";
import Catalog from "../Catalog/Catalog";
import Logo from "../UI/Logo/Logo";
import ButtonLogIn from "../UI/ButtonLogIn/ButtonLogIn";

const HeaderFixed = (props) => {
  const count = props.cart.reduce((sum, item) => {
    return item.count + sum
  }, 0)

  const [lastScrollY, setLastScrollY] = useState(0)
  const [popup, setPopup] = useState(false)
  const [burgerActive, setBurgerActive] = useState(false)
  // const isLogin = () => {
  //   console.log('JSON.parse(localStorage.getItem("TOKEN")', JSON.parse(localStorage.getItem("TOKEN")))
  //   return localStorage.getItem('isLogin') === 'true'
  // }

  useEffect(() => {
    const handleScroll = () => {
      setLastScrollY(window.scrollY)
    };
    window.addEventListener('scroll', handleScroll);
    return () => window.removeEventListener('scroll', handleScroll);
  })

  return (
    <>
      <div className={'HeaderFixed ' + (lastScrollY > 33 ? 'HeaderFixed--active' : '')}>
        <div className='wrapper HeaderFixed__wrapper'>
          <div className='HeaderFixed__leftBlock'>
            <div className='HeaderFixed__catalog'>
              <Burger isActive={burgerActive} onClick={() => setBurgerActive(!burgerActive)}/>
              {props.catalog && <Catalog isActive={burgerActive}
                                         onClick={() => setBurgerActive(!burgerActive)}
                                         data={props.catalog.child}
              />}
            </div>
            <Logo/>
          </div>

          {/*<SearchPanel/>*/}

          <div className='HeaderFixed__rightblock'>

            <ButtonLogIn
              onClick={() => {
                if (props.TOKEN) {
                  props.history.push('/profile/')
                  window.scroll(0, 0)
                } else setPopup(true)
              }}
            >{props.TOKEN ? 'личный кабинет' : 'войти'}
            </ButtonLogIn>

            <NavLink to="/cart/" className='HeaderFixed__cart'>
              <div className='HeaderFixed__cartImgBox'>
                <img src={iconCart} alt="корзина" className='HeaderFixed__cartImg'/>
              </div>
              <span className='HeaderFixed__cartCount'>{count}</span>
            </NavLink>
          </div>
        </div>
        {lastScrollY > 400 && <ButtonTopScroll/>}
        <PopupLogin active={popup}
                    onClick={() => setPopup(false)}
        />
      </div>
      <div style={{width: '100%', height: 64, display: lastScrollY > 33 ? 'flex' : 'none'}}> </div>
    </>
  )
}

const mapStateToProps = ({cart, TOKEN, catalog}) => {
  return {cart, TOKEN, catalog}
}

const mapDispatchToProps = (dispatch) => {
  return {
    rewriteCart: (item) => dispatch(rewriteCart(item)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(HeaderFixed))

