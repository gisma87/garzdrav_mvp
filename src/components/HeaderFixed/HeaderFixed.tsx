import React, {useEffect, useState} from "react"
import './HeaderFixed.scss'
import {NavLink, RouteComponentProps, withRouter} from "react-router-dom";
import iconCart from '../../img/icon/cartIconSmall.png'
import ButtonTopScroll from "../UI/ButtonTopScroll";
import {openPopupLogin, rewriteCart} from "../../actions";
import {connect} from "react-redux";
import Burger from "../UI/Burger/Burger";
import Catalog from "../Catalog/Catalog";
import Logo from "../UI/Logo/Logo";
import ButtonLogIn from "../UI/ButtonLogIn/ButtonLogIn";
import ErrorBoundary from "../ErrorBoundary/ErrorBoundary";
import {CartItemType, CategoryElement} from "../../types";
import {StateType} from "../../store";
import {ThunkDispatch} from "redux-thunk";

type MapDispatchPropsType = {
  openPopupLogin(): void,
  rewriteCart(cart: CartItemType[]): void
}

type MapStatePropsType = {
  cart: CartItemType[],
  catalog: null | CategoryElement,
  TOKEN: null | { accessToken: string, refreshToken: string }
}

type Props = RouteComponentProps & MapStatePropsType & MapDispatchPropsType;

const HeaderFixed: React.FC<Props> = (props) => {
  const count = props.cart.reduce((sum, item) => {
    return item.count + sum
  }, 0)

  const [lastScrollY, setLastScrollY] = useState(0)
  const [burgerActive, setBurgerActive] = useState(false)

  useEffect(() => {
    const handleScroll = () => {
      setLastScrollY(window.scrollY)
    };
    window.addEventListener('scroll', handleScroll);
    return () => window.removeEventListener('scroll', handleScroll);
  })

  useEffect(() => {
    if (burgerActive) {
      document.body.style.overflow = 'hidden'
    } else {
      document.body.style.overflow = 'auto'
    }
  }, [burgerActive])

  return (
    <ErrorBoundary>
      <div className='HeaderFixed'>
        <div className='wrapper HeaderFixed__wrapper'>
          <div className='HeaderFixed__leftBlock'>
            <div className='HeaderFixed__catalog'>
              <Burger isActive={burgerActive} onClick={() => setBurgerActive(!burgerActive)}/>
              {props.catalog && <Catalog isActive={burgerActive}
                                         onClick={() => setBurgerActive(!burgerActive)}
                                         data={props.catalog.child}
              />}
            </div>
            <Logo/>
          </div>

          <p className='test'>Тестирование</p>

          <div className='HeaderFixed__rightblock'>

            <ButtonLogIn
              onClick={() => {
                if (props.TOKEN) {
                  props.history.push('/profile/')
                  window.scroll(0, 0)
                } else props.openPopupLogin()
              }}
            >{props.TOKEN ? 'личный кабинет' : 'войти'}
            </ButtonLogIn>

            <NavLink to="/cart/" className='HeaderFixed__cart'>
              <div className='HeaderFixed__cartImgBox'>
                <img src={iconCart} alt="корзина" className='HeaderFixed__cartImg'/>
              </div>
              <span className='HeaderFixed__cartCount'>{count}</span>
            </NavLink>
          </div>
        </div>
        {lastScrollY > 400 && <ButtonTopScroll/>}
      </div>
    </ErrorBoundary>
  )
}

const mapStateToProps = ({cart, TOKEN, catalog}: StateType) => {
  return {cart, TOKEN, catalog}
}

const mapDispatchToProps = (dispatch: ThunkDispatch<StateType, {}, any>) => {
  return {
    openPopupLogin: () => dispatch(openPopupLogin()),
    rewriteCart: (cart: CartItemType[]) => dispatch(rewriteCart(cart)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(HeaderFixed))

