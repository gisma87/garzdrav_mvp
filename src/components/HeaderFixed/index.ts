import HeaderFixed from "./HeaderFixed";
export default HeaderFixed