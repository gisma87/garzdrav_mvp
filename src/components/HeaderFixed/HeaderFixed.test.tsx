import {render, screen} from "@testing-library/react";
import {Provider} from "react-redux";
import store from "../../store";
import {BrowserRouter} from "react-router-dom";
import React from "react";
import HeaderFixed from "./HeaderFixed";
import '@testing-library/jest-dom'
import {addedToCart, allItemRemovedFromCart, itemRemovedFromCart} from "../../actions";


describe('HeaderFixed', () => {
    test('HeaderFixed snapshot', () => {
        const {asFragment, container} = render(
            <Provider store={store}>
                <BrowserRouter basename="/">
                    <HeaderFixed/>
                </BrowserRouter>
            </Provider>
        )

        const cartCount = container.getElementsByClassName('HeaderFixed__cartCount')

        expect(asFragment()).toMatchSnapshot()
        expect(cartCount[0].textContent).toBe("0")

        store.dispatch(addedToCart('idItemCart_01'))
        expect(cartCount[0].textContent).toBe("1")

        store.dispatch(addedToCart('idItemCart_01'))
        expect(cartCount[0].textContent).toBe("2")

        store.dispatch(addedToCart('idItemCart_01'))
        store.dispatch(addedToCart('idItemCart_01'))
        store.dispatch(addedToCart('idItemCart_01'))
        expect(cartCount[0].textContent).toBe("5")

        store.dispatch(itemRemovedFromCart('idItemCart_01'))
        store.dispatch(itemRemovedFromCart('idItemCart_01'))
        expect(cartCount[0].textContent).toBe("3")

        store.dispatch(allItemRemovedFromCart('idItemCart_01'))
        expect(cartCount[0].textContent).toBe("0")

        store.dispatch(itemRemovedFromCart('idItemCart_01'))
        store.dispatch(allItemRemovedFromCart('idItemCart_01'))
        expect(cartCount[0].textContent).toBe("0")
    })
})