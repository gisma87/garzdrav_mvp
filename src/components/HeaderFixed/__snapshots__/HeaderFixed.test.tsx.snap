// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderFixed HeaderFixed snapshot 1`] = `
<DocumentFragment>
  <div
    class="HeaderFixed"
  >
    <div
      class="wrapper HeaderFixed__wrapper"
    >
      <div
        class="HeaderFixed__leftBlock"
      >
        <div
          class="HeaderFixed__catalog"
        >
          <div
            class="burger "
          >
            <div
              class="burger__span"
            >
              <span />
            </div>
          </div>
        </div>
        <a
          aria-current="page"
          class="Logo active"
          href="/"
        >
          <img
            alt="логотип"
            class="Logo__img"
            src="iconALsmall.png"
          />
          <div
            class="Logo__textContent"
          >
            <p
              class="Logo__legko"
            >
              легко
            </p>
            <div
              class="Logo__text"
            >
              <p
                class="Logo__internet"
              >
                интернет
              </p>
              <p>
                аптека
              </p>
            </div>
          </div>
        </a>
      </div>
      <p
        class="test"
      >
        Тестирование
      </p>
      <div
        class="HeaderFixed__rightblock"
      >
        <button
          class="ButtonLogIn"
        >
          <img
            alt="Вход"
            class="ButtonLogIn__img"
            src="LogInSmall.png"
          />
          <p
            class="ButtonLogIn__text"
          >
            войти
          </p>
        </button>
        <a
          class="HeaderFixed__cart"
          href="/cart/"
        >
          <div
            class="HeaderFixed__cartImgBox"
          >
            <img
              alt="корзина"
              class="HeaderFixed__cartImg"
              src="cartIconSmall.png"
            />
          </div>
          <span
            class="HeaderFixed__cartCount"
          >
            0
          </span>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
